arrayfire/array.py

```diff
@@ -1185,20 +1185,35 @@ def to_list(self, row_major=False):
         ct_array, shape = self.to_ctype(row_major, True)
         return _ctype_to_lists(ct_array, len(shape) - 1, shape)
 
-    def __repr__(self):
+    def to_string(self):
```

Comment:
We could do:

* introduce a constant on top of this file that defines a dim limit separating arrays to two category: small and large
* modify the proposed `to_string` method to accept an argument: `display_dims_limit` which is a tuple telling us a dimensions limit of the array after which we won't show its data in the string representation. If this argument is None then we use the above default.
* add an `__str__` method that simply calls `self.to_string()` (with the default limit)

In this way we wont break print on small arrays, and we have a method to get the a full array in string if that's what the user want.